roles/container-engine/docker/defaults/main.yml

```diff
@@ -43,7 +43,7 @@ docker_debian_repo_base_url: "https://download.docker.com/linux/debian"
 docker_debian_repo_gpgkey: 'https://download.docker.com/linux/debian/gpg'
 docker_bin_dir: "/usr/bin"
 # CentOS/RedHat Extras repo
-extras_rh_repo_base_url: "http://mirror.centos.org/centos/$releasever/extras/$basearch/"
+extras_rh_repo_base_url: "http://mirror.centos.org/centos/7/extras/$basearch/"
```

Comment:
Why fixing release to 7 ?

Reply:
Unfortunately, across RHEL hosts the **$releasever** value gets assigned as `7Server`, hence the change.

I specifically hardcoded it to `7` to match it with the CentOS RPM GPG key.

```
extras_rh_repo_gpgkey: "http://mirror.centos.org/centos/RPM-GPG-KEY-CentOS-7"
```

As any change to make this work with RHEL 7/8 OS's will likely impact CentOS and Oracle Linux hosts as well, I decided to keep the changes limited in scope for this PR.

There needs to be some extra work via a different PR to change the RH Extras parameters related to installing docker/containerd to work across all RHEL, CentOS and Oracle Linux 7/8 hosts.

e.g.
extras_rh_repo_base_url
extras_rh_repo_gpgkey

Interestingly, the Docker download URLs still only have version `7` available for the `docker-ce` packages.
https://download.docker.com/linux/centos/

So the following parameters don't need updating just yet to take into account various RHEL, CentOS and Oracle Linux 7/8 hosts.

docker_rh_repo_base_url
docker_rh_repo_gpgkey

Should I try and put in place something similiar to the following Fedora parameter to get this to work across the different Red Hat OS versions?

```
docker_fedora_release: "{{ '31' if (ansible_distribution_major_version | int) > 31 else ansible_distribution_major_version }}"
```

Reply:
I think I've answered my own question...

I'll see if I can come up with a solution that works with all Red Hat 7/8 OS distros.